### PR TITLE
Update LUT routine

### PR DESCRIPTION
**Pull request by bigfooted**
## Proposed Changes
*Give a brief overview of your contribution here in a few sentences.*
 This is a small update of the lookup table routines to prepare for the implementation of the flamelet model. A bug was fixed and the header file definition changed (keywords in sentences are capitalized). Some small textual changes were made.

## Related Work
*Resolve any issues (bug fix or feature request), note any related PRs, or mention interactions with the work of others, if any.*
related to the upcoming PR for feature_new_flamelet

## PR Checklist
*Put an X by all that apply. You can fill this out after submitting the PR. If you have any questions, don't hesitate to ask! We want to help. These are a guide for you to know what the reviewers will be looking for in your contribution.*

- [x] I am submitting my contribution to the develop branch.
- [x] My contribution generates no new compiler warnings (try with --warnlevel=3 when using meson).
- [x] My contribution is commented and consistent with SU2 style (https://su2code.github.io/docs_v7/Style-Guide/).
- [ ] I have added a test case that demonstrates my contribution, if necessary.
- [ ] I have updated appropriate documentation (Tutorials, Docs Page, config_template.cpp), if necessary.

### CODE REVIEW

**Comment by bigfooted** [comment]
@danielmayer I've created a separate PR for the changes in the existing LUT files. Maybe we can replace these lines with a function:
```
   if (!line.empty() && (line[line.length()-1] == '\n' || line[line.length()-1] == '\r' )) {
      line.erase(line.length()-1);
    }
```

File: UnitTests/Common/containers/lookuptable.drg [hunk 1]
```diff
@@ -4,28 +4,28 @@ Dragon library
 [version]
 1.0.0
```
Comment (bigfooted) [inline UnitTests/Common/containers/lookuptable.drg:5]:
```suggestion
1.0.1
```

File: Common/include/containers/CFileReaderLUT.hpp [hunk 2]
```diff
@@ -83,5 +83,12 @@ class CFileReaderLUT {
 
   inline const std::vector<unsigned long>& GetHull() const { return hull; };
 
+  inline std::string GetStrippedLine(std::string line) {
+    if (!line.empty() && (line[line.length()-1] == '\n' || line[line.length()-1] == '\r' )) {
```
Comment (pcarruscag) [inline Common/include/containers/CFileReaderLUT.hpp:87]:
```suggestion
    if (!line.empty() && (line.back() == '\n' || line.back() == '\r' )) {
```

File: Common/include/containers/CFileReaderLUT.hpp [hunk 3]
```diff
@@ -83,5 +83,12 @@ class CFileReaderLUT {
 
   inline const std::vector<unsigned long>& GetHull() const { return hull; };
 
+  inline std::string GetStrippedLine(std::string line) {
```
Comment (pcarruscag) [inline Common/include/containers/CFileReaderLUT.hpp:86]:
```suggestion
  inline static std::string StripLine(std::string line) {
```

File: Common/include/containers/CFileReaderLUT.hpp (hunk 2 above)
Comment (bigfooted) [reply Common/include/containers/CFileReaderLUT.hpp]: @pcarruscag I was just working on a replacement, using find_last_not_of

File: Common/include/containers/CFileReaderLUT.hpp [hunk 4]
```diff
@@ -83,5 +83,11 @@ class CFileReaderLUT {
 
   inline const std::vector<unsigned long>& GetHull() const { return hull; };
 
+  /*--- strip trailing characters ---*/
+  inline static std::string StripLine(std::string line) {
+    size_t end = line.find_last_not_of(" \n\r\t\f\v");
```
Comment (pcarruscag) [inline Common/include/containers/CFileReaderLUT.hpp:88]: :+1:

File: Common/src/containers/CFileReaderLUT.cpp [hunk 5]
```diff
@@ -238,15 +240,67 @@ void CFileReaderLUT::ReadRawDRG(const string& file_name) {
...
-  string line;
-  getline(*file_stream, line);
+/**
+ * @brief Searches for the position of flag in file_stream and 
+ *        sets the stream position of file_stream to that position.
+ * 
+ * @param file_stream 
+ * @param current_line 
+ * @param flag 
+ */
```
Comment (pcarruscag) [inline Common/src/containers/CFileReaderLUT.cpp:250]: Please use the style we have in other parts of the code `\param[in]` `\brief` and there is a `!` missing somewhere.

File: Common/src/containers/CFileReaderLUT.cpp [hunk 6]
```diff
@@ -238,15 +240,67 @@ void CFileReaderLUT::ReadRawDRG(const string& file_name) {
...
+void CFileReaderLUT::SkipToFlag(ifstream* file_stream, const string& current_line, const string& flag) {
```
Comment (pcarruscag) [inline Common/src/containers/CFileReaderLUT.cpp:251]: These helper methods should be either const, or static, or free functions in an anonymous namespace.
For performance they should be inline.
Please pass file_stream by reference

File: Common/src/containers/CTrapezoidalMap.cpp [hunk 7]
```diff
@@ -174,7 +175,9 @@ pair<unsigned long, unsigned long> CTrapezoidalMap::GetBand(su2double val_x) {
 
   std::pair<std::vector<su2double>::iterator,std::vector<su2double>::iterator> bounds;
   bounds = std::equal_range (unique_bands_x.begin(), unique_bands_x.end(), val_x);
-  i_up =  bounds.first - unique_bands_x.begin();
+
+  /*--- if upper bound = 0, then use the range [0,1] ---*/
+  i_up =  max((unsigned long)(1), (unsigned long)(bounds.first - unique_bands_x.begin()));
```
Comment (pcarruscag) [inline Common/src/containers/CTrapezoidalMap.cpp:180]: Don't use c-style casts please.
```suggestion
  i_up =  max<unsigned long>(1, bounds.first - unique_bands_x.begin());
```

File: Common/src/containers/CFileReaderLUT.cpp [hunk 8]
```diff
@@ -59,106 +57,104 @@ void CFileReaderLUT::ReadRawDRG(const string& file_name) {
...
     }
 
-    // check if end of header is reached
-    if (line.compare("</header>") == 0) eoHeader = true;
+    /*--- check if end of header is reached ---*/
+    if (line.compare("</Header>") == 0) eoHeader = true;
   }
 
-  // check version_lut
+  /*--- check version_lut ---*/
```
Comment (pcarruscag) [inline Common/src/containers/CFileReaderLUT.cpp:110]: Thank you for making these consistent :+1:

File: Common/src/containers/CFileReaderLUT.cpp (hunk 6 above)
Comment (bigfooted) [reply Common/src/containers/CFileReaderLUT.cpp]: OK, I think making them const makes the most sense, it is also what we do with the other functions. Are you asking to make them inline? They only go through the relatively small header file, so there are no performance issues.

File: Common/src/containers/CFileReaderLUT.cpp [hunk 9]
```diff
@@ -231,22 +233,62 @@ void CFileReaderLUT::ReadRawDRG(const string& file_name) {
...
+    return_value = GetStrippedLine(file_stream, line);
+  }
+
+  /*--- return true if line is not empty, else return false ---*/
+  return return_value;
 }
+
+/*! \brief Extracts characters from file_stream, removes trailing control characters,
+ *         and stores them into line.
+ */
```
Comment (pcarruscag) [inline Common/src/containers/CFileReaderLUT.cpp:280]: Btw it is usually better to put the doxygen comment on the declaration hpp, but it's fine

File: Common/src/containers/CFileReaderLUT.cpp (hunk 6 above)
Comment (danielmayer) [reply Common/src/containers/CFileReaderLUT.cpp]: @pcarruscag Thank you for your comments, Pedro. I think functions defined in classes are implicitly inline anyways, aren't they? Happy to add some inline specifiers to be sure, though!

Comment (pcarruscag) [reply Common/src/containers/CFileReaderLUT.cpp]: They are implicitly inline if you define them in the body of the class, if you declare them in the body of the class and define them in a cpp they are not inline.
However, maybe the compiler can still inline within the cpp where they are defined (they will def. not be inlined in other places that only see the declaration of the function but not its definition, e.g. other cpps, unless you turn on link-time optimizations).
But like Nijso said, probably not important here.

File: Common/src/containers/CFileReaderLUT.cpp (hunk 9 above)
Comment (bigfooted) [reply Common/src/containers/CFileReaderLUT.cpp]: Thanks, I've moved it to the hpp files. When it goes through the regression tests, I'll merge.

File: Common/src/containers/CFileReaderLUT.cpp (hunk 6 above)
Comment (danielmayer) [reply Common/src/containers/CFileReaderLUT.cpp]: Cool, thank you!

**Pull request merged**